packages/app/src/public-types.ts

```diff
@@ -139,6 +182,65 @@ export interface FirebaseAppSettings {
   automaticDataCollectionEnabled?: boolean;
 }
 
+/**
+ * @public
+ *
+ * Configuration options given to {@link (initializeServerApp:1) | initializeServerApp()}
+ */
+export interface FirebaseServerAppSettings extends FirebaseAppSettings {
+  /**
+   * An optional Auth ID token used to resume a signed in user session from a client
+   * runtime environment.
+   *
+   * If provided, the `FirebaseServerApp` works to validate the token even before
+   * `Auth` is initialized. The result of the validation can be queried via
+   * {@link (FirebaseServerApp.authIdTokenVerified()}. Awaiting the `Promise` returned by
+   * {@link (FirebaseServerApp.authIdTokenVerified()} is highly recommended if an `authIdToken`
+   * token is provided.
+   *
+   * Invoking `getAuth` with a `FirebaseServerApp` configured with a validated `authIdToken`
+   * causes an automatic attempt to sign in the user that the `authIdToken` represents. The token
+   * needs to have been recently minted for this operation to succeed.
+   *
+   * If the token fails local verification, or if the Auth service has failed to validate it when
+   * the Auth SDK is initialized, then a warning is logged to the console and the Auth SDK will not
+   * sign in a user on initalization.
+   *
+   * If a user is successfully signed in, then the Auth instance's `onAuthStateChanged` callback
+   * is invoked with the `User` object as per standard Auth flows. However, `User` objects
+   * created via an `authIdToken` do not have a refresh token. Attempted `refreshToken`
+   * operations fail.
+   */
+  authIdToken?: string;
+
+  /**
+   * An optional object. If provided, the Firebase SDK uses a `FinalizationRegistry`
+   * object to monitor the garbage collection status of the provided object. The
+   * Firebase SDK releases its refrence on the `FirebaseServerApp` instance when the
+   * provided `releaseOnDeref` object is garbage collected.
+   *
+   * The intent of this field is to help reduce memory overhead for cloud functions.
```

Comment:
Suggested revision; may need your input for accuracy:

"   * If your app runs functions on the server, you can use this field to reduce memory overhead for those functions.
   * If provided, an app running in a SSR pass does not need to perform
   * `FirebaseServerApp` cleanup, so long as the reference object is deleted (by falling out of
   * SSR scope, for instance.)"

Reply:
I think we shouldn't say "If your app runs functions on the server..." since this is named `FirebaseServerApp` already. How about this:

```
You can use this field to reduce memory management overhead for your application.
If provided, an app running in a SSR pass does not need to perform
`FirebaseServerApp` cleanup, so long as the reference object is deleted (by falling out of
 SSR scope, for instance.)
 ```

Reply:
Sounds good!  What we don't want is any avoidable mangling of "Cloud Functions" the product or "cloud function" the actual code entity, which should never be preceded by lowercase "cloud."  A function is a function, whether it runs on Cloud Functions or wherever.

Reply:
Yeah. The previous cloud functions was the concept, not the product. But now there's no confusion at all!